CONTRIBUTING.md

```diff
@@ -1,331 +1,444 @@
 
-Once code review starts on your PR, do not rebase nor squash your branch as it makes it
-difficult to review the new changes. Only if there is a need, sync your branch by merging
-the base branch into yours. Don't worry about merge commits messing up your tree as
-the final merge process squashes all commits into one, with the visible commit message (first
-line) being the PR title + PR index and description being the PR's first comment.
+### Maintaining open PRs
 
-Once your PR gets the `lgtm/done` label, don't worry about keeping it up-to-date or breaking
-builds (unless there's a merge conflict or a request is made by a maintainer to make
-modifications). It is the maintainer team's responsibility from this point to get it merged.
+The moment you create a non-draft PR or the moment you convert a draft PR to a non-draft PR is the moment code review starts for it. \
+Once that happens, do not rebase or squash your branch anymore as it makes it difficult to review the new changes. \
+Merge the base branch into your branch only when you really need to, i.e. because of conflicting changes in the mean time. \
+This reduces unnecessary CI runs. \
+Don't worry about merge commits messing up your commit history as every PR will be squash merged. \
+This means that all changes are joined into a single new commit whose message is as described below.
 
-## Styleguide
 
-For imports you should use the following format (*without* the comments)
+### Getting PRs merged
 
-```go
-import (
-  // stdlib
-  "fmt"
-  "math"
+Changes to Gitea must be reviewed before they are accepted — no matter who
+makes the change, even if they are an owner or a maintainer. \
+The only exception are critical bugs that prevent Gitea from being compiled or started. \
+Specifically, we require two approvals from maintainers for every PR. \
+Once this criteria has been met, your PR receives the `lgtm/done` label. \
+From this point on, your only responsibility is to fix merge conflicts or respond to/implement requests by maintainers. \
+It is the responsibility of the maintainers from this point to get your PR merged.
 
-  // local packages
-  "code.gitea.io/gitea/models"
-  "code.gitea.io/sdk/gitea"
+If a PR has the `lgtm/done` label and there are no open discussions or merge conflicts anymore, any maintainer can add the `reviewed/wait-merge` label. \
+This label means that the PR is part of the merge queue and will be merged as soon as possible. \
+The merge queue will be cleared in the order of the list below:
 
-  // external packages
-  "github.com/foo/bar"
-  "gopkg.io/baz.v1"
-)
-```
+<https://github.com/go-gitea/gitea/pulls?q=is%3Apr+label%3Areviewed%2Fwait-merge+sort%3Acreated-asc+is%3Aopen>
 
-## Design guideline
+Gitea uses it's own tool, the <https://github.com/GiteaBot/gitea-backporter> to automate parts of the review process. \
+This tool does the things listed below automatically:
+- create a backport PR if needed once the initial PR was merged
```

Comment:
At _attempts_ to do this, but is not always successful, e.g. when conflicts occur
Do we want to mention that here?

Reply:
I think maybe we could improve the whole documentation detail in the docs section `Contributing` but not make this file bloat. About the robot, they are changed frequently recently.